packages/core/contracts-ovm/insured-bridge/implementation/BridgePool.sol

```diff
@@ -297,6 +350,26 @@ contract BridgePool is Testable {
         emit RelayDisputed(depositHash, keccak256(ancillaryData));
     }
 
+    /************************************
+     *           View FUNCTIONS         *
+     ************************************/
+
+    /**
+     * @notice Computes the exchange rate between LP tokens and L1Tokens. Used when adding/removing liquidity.
+     */
+    function exchangeRateCurrent() public view returns (uint256) {
+        if (totalSupply() == 0) return 1e18; //initial rate is 1 pre any mint action.
+
+        // Consider a naive rate implementation. This acts like a step function, increasing when funds hit L1 from the
+        // canonical bridge. TODO: update with a more elaborate technique that pays out gradually over the 1 week loan.
```

Comment:
Is this function where the fee-payment is defined? This function seems unrelated to that, but maybe I'm misunderstanding?

Reply:
this function defines the exchange rate between LP tokens and the underlying L1Tokens. It is indeed where the fee payment happens. In this implementation, it is a simple step function that increments at the conclusion of transfers. it's naive and does not follow any of the latest thinking we have with the exponential fees. it's done like this so we don't need to define complex fee payment within this PR.